src/System.Net.NameResolution/tests/PalTests/NameResolutionPalTests.cs

```diff
@@ -45,6 +44,14 @@ public void TryGetAddrInfo_HostName()
             IPHostEntry hostEntry;
             int nativeErrorCode;
             SocketError error = NameResolutionPal.TryGetAddrInfo(hostName, out hostEntry, out nativeErrorCode);
+            if (error == SocketError.HostNotFound)
+            {
+                // On Unix, we are not guaranteed to be able to resove the local host. The ability to do so depends on the 
+                // machine configurations, which varies by distro and is often inconsistent.
+                Assert.True(RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX));
```

Comment:
Not: I wonder if it would be "better" to move OS check into the if-condition. If we ever get HostNotFound on Windows, it should ideally fail on the `Assert.Equal(SocketError.Succes, error);` line.
If this is established pattern in other tests, I'm fine leaving it as is.

Reply:
I would rather the OS checks in the if-condition.  This is usually how we do OS platform-specific tests using 'ifs'.  It makes the code easier to read if we see things like this:

```c#
if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
{
    // Test setup and assertions specific to Linux
}
else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
{
    // Test setup and assertions specific to OSX
}

// ...
```

Reply:
This is really not specific to OS but given environment. Linux _can_ resolve the names if configured as such. Add it will in most cases.

Reply:
Do you think it's better to do things this way, or to set up a capability as mentioned by @wfurt below?

Reply:
Capabilities are not used for this kind of thing.  This is basically a platform difference in behavior that you want to capture in a test.

Reply:
Made this change. I agree that it's an improvement over the check I initially added -- it should make failures a lot more clear.